src/client/Microsoft.Identity.Client/AppConfig/PopAuthenticationConfiguration.cs

```diff
@@ -3,46 +3,99 @@
         /// </summary>
-        /// <param name="requestUri">An Uri to the protected resource which requires a PoP token. The PoP token will be cryptographically bound to the request uri</param>
-        public PoPAuthenticationConfiguration(Uri requestUri)
+        /// <remarks>
+        /// See https://datatracker.ietf.org/doc/html/draft-ietf-oauth-signed-http-request-03#page-3 for details about signed HTTP requests.
+        /// </remarks>
+        public PoPAuthenticationConfiguration()
         {
             ConfidentialClientApplication.GuardMobileFrameworks();
-            RequestUri = requestUri ?? throw new ArgumentNullException(nameof(requestUri));
         }
 
         /// <summary>
-        /// An Uri to the protected resource which requires a PoP token. The PoP token will be cryptographically bound to the request uri (to the host and query parts of it)
+        /// Creates a configuration using the default key management, and which binds all the details of the HttpRequestMessage.
         /// </summary>
-        public Uri RequestUri { get; }
+        /// <remarks>
+        /// Currently only the HtppMethod (m), UrlHost (u) and UrlPath (p) are used to create the signed HTTP request - see https://datatracker.ietf.org/doc/html/draft-ietf-oauth-signed-http-request-03#page-3
+        /// </remarks>
+        public PoPAuthenticationConfiguration(HttpRequestMessage httpRequestMessage)
+        {
+            if (httpRequestMessage == null)
+                throw new ArgumentNullException(nameof(httpRequestMessage));
+
+            HttpMethod = httpRequestMessage.Method;
+            HttpHost = httpRequestMessage.RequestUri.Authority; // this includes the optional port
+            HttpPath = httpRequestMessage.RequestUri.AbsolutePath; 
+        }
+
+      
+
+        /// <summary>
+        /// Creates a configuration using the default key management, and which binds only the Uri part of the HTTP request.
+        /// </summary>
+        /// <remarks>
+        /// The UrlHost (u) and UrlPath (p) are used to create the signed HTTP request - see https://datatracker.ietf.org/doc/html/draft-ietf-oauth-signed-http-request-03#page-3
+        /// </remarks>
+        public PoPAuthenticationConfiguration(Uri requestUri)
+        {
+            if (requestUri == null)
+                throw new ArgumentNullException(nameof(requestUri));
+
+            HttpHost = requestUri.Authority;
+            HttpPath = requestUri.AbsolutePath;
+        }
 
         /// <summary>
         /// The HTTP method ("GET", "POST" etc.) method that will be bound to the token. Leave null and the POP token will not be bound to the method.
+        /// Corresponds to the "m" part of the a signed HTTP request. Optional.
         /// </summary>
+        /// <remarks>
+        /// See https://datatracker.ietf.org/doc/html/draft-ietf-oauth-signed-http-request-03#section-3
+        /// </remarks>
         public HttpMethod HttpMethod { get; set; }
```

Comment:
Would we want to have Get as a default method?

Reply:
This class will evolve to have all HTTP request components: q (query), h (headers), b (body). All of these claims are optional, so it seems a bit unnatural to have m (method) take a default value.